Add support for EZValo Smart Desk Lamp

aminorjourney opened the pull request:
Added support for https://www.ezvaloledlight.com/product-category/smart-desk-lamp/, including: 

• Power On/Off of the light
• Dimmer
• Color Temperature
• "Auto" Color/Temperature mode
• Timer (for shutdown) - limited to 4 hours (240 minutes)

aminorjourney committed:
Add support for EZValo Smart Desk Lamp

Added support for https://www.ezvaloledlight.com/product-category/smart-desk-lamp/, including: 

• Power On/Off of the light
• Dimmer
• Color Temperature
• "Auto" Color/Temperature mode
• Timer (for shutdown) - limited to 4 hours (240 minutes)

aminorjourney committed:
Rename Auto Brightness and Color Mode to switch

Fixed a bug that wasn't present when submitted - fixing now.

make-all committed:
Refactor Ezvalo Smart Desk Lamp YAML configuration

Updated the name and formatting for the Ezvalo Smart Desk Lamp configuration.
Move to the correct folder

PR #4069

make-all commented on custom_components/tuya_local/devices/ezvalo_smart_desklamp.yaml:
```diff
@@ -0,0 +1,48 @@
+              max: 6500
+              
+  - entity: switch
+    name: Auto mode
```
Is this some light sensing or sunrise/sunset tracking mode that changes color temperature and brightness according to time of day? Or something else? It may be better to choose a name that more precisely describes the feature (without getting too wordy)

aminorjourney replied:
Sorry for the delay. 

The auto function controls brightness and color temperature based on the ambient light in the room.  Do you have any specific request for naming that feature?  

I would perhaps call it "Auto Ambient Light"?  I will look into the ID shortly!

make-all committed:
Fix formatting in ezvalo_smart_desklamp.yaml

make-all committed:
feat (ezvalo_smart_desklamp): Rename 'Auto mode' to 'Auto dimming'

Based on description at https://www.amazon.com/EZVALO-Dimmable-Eye-Caring-Lighting-Adjustable/dp/B0BQMNMYM2

PR #4069

mapeol commented:
Hi!
This is excellent work, but I can't get HA to recognise the colour temperature.
Can you help me?
